code/datums/explosion.dm

```diff
@@ -14,6 +14,7 @@ GLOBAL_LIST_EMPTY(explosions)
 
 /datum/explosion
 	var/explosion_id
+	var/atom/explosion_source
```

Comment:
This is gonna cause GC errors because this reference will prevent the source from deleting and it stays forever

Reply:
is that right? the explosion datum itself is qdeleted after the explosion finishes, which means that assuming nothing references the explosion datum elsewhere, the explosion datum will GC and the reference to the source will disappear alongside... in a normal programming language...???

Should I null the reference in Destroy anyway?

Reply:
Nevermind I'm full of shit